odoo/addons/base/models/ir_http.py

```diff
@@ -396,6 +396,12 @@ def _binary_record_content(
         if not content:
             if model == 'ir.attachment':
```

Comment:
this line seems a bit naive, it's the combination of both `ir.attachment` and `datas` that needs to be replaced by `raw`

Reply:
I think that it's like that just to avoid the `else` clause (in which we do a b64decode) if we request the `raw` field

Reply:
Yes but this means that the fieldname is no longer relevant. If you go to `/web/content/ir.attachment/<id>/name` you will get the content as well. Let say we add a new binary field on ir.attachment (e.g. "old_content"), it would be ignored. Not a big deal but still a bug.

Reply:
@mart-e Since it's not related to this PR, I will make another one for this. Do you want it on master or 15.2?